src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,117 @@
+package duke;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Storage class to handle loading and saving of tasks to a text file.
+ */
+public class Storage {
+    private String filePath;
+
+    /**
+     * Initializes a Storage object with file path to text file.
+     *
+     * @param filePath the path of the file to save the tasks to
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads the tasks from the file, processing of text to task is done in parseTasksFromString.
+     *
+     * @return the list of tasks loaded from the file.
+     * @throws IOException if there is an error reading the file.
+     * @throws DukeException if the file could not be created.
+     */
+    public ArrayList<Task> loadTasks() throws IOException, DukeException {
+        ArrayList<Task> tasksList = new ArrayList<Task>();
+
+        File file = new File(filePath);
+        if (!file.exists()) {
+            boolean isCreated = false;
+            isCreated = file.getParentFile().mkdirs();
+            isCreated = file.createNewFile();
+
+            if (!isCreated) {
+                throw new DukeException("File could not be created.");
+            }
+        } else {
+            Scanner s = new Scanner(file);
+            while (s.hasNext()) {
+                String line = s.nextLine();
+                Task task = parseTasksFromString(line);
+                tasksList.add(task);
+            }
+            s.close();
+        }
+
+        return tasksList;
+    }
+
+    private Task parseTasksFromString(String line) throws DukeException {
+        String[] split = line.split(" \\| ");
```

Comment:
Perhaps split could be renamed to splits or parsedInputs to avoid confusion to a single-valued variable?